src/SFML/Network/TcpSocket.cpp

```diff
@@ -242,11 +242,14 @@ Socket::Status TcpSocket::send(const void* data, std::size_t size, std::size_t&
     }
 
     // Loop until every byte has been sent
-    int result = 0;
-    for (sent = 0; sent < size; sent += result)
+    ssize_t result = 0;
+    for (sent = 0; sent < size; sent += static_cast<std::size_t>(result))
     {
+        #pragma GCC diagnostic push
+        #pragma GCC diagnostic ignored "-Wuseless-cast"
```

Comment:
For Clang this is not needed, but GCC will complain about useless cast, but if you keep it, then Clang will complain about not knowing this warning, so it can't ignore it. So we can't use it, but also can't not use it.

I guess at this point using a `#if define(unix/windows)` is probably better...